internal/resource.go

```diff
@@ -109,27 +113,56 @@ func (l *Resource) Download(dir string, mode os.FileMode, ctx context.Context) e
 			localName = path.Base(u)
 		}
 		resPath := filepath.Join(dir, localName)
-		err = os.Rename(lpath, resPath)
+
+		// Check existing file first
+		if _, err := os.Stat(resPath); err == nil {
```

Comment:
This code is a bit wasteful: you're doing `os.Stat` multiple times. It's not *too* bad of course but you could structure this better to be more efficient *and* more clear for the reader.